third-party/crowdsec/src/main/java/org/apache/james/CrowdsecImapConnectionCheck.java

```diff
@@ -0,0 +1,71 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.net.InetSocketAddress;
+
+import javax.inject.Inject;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.apache.james.imap.api.ConnectionCheck;
+import org.apache.james.model.CrowdsecClientConfiguration;
+import org.apache.james.model.CrowdsecDecision;
+import org.apache.james.model.CrowdsecHttpClient;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CrowdsecImapConnectionCheck implements ConnectionCheck {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrowdsecImapConnectionCheck.class);
+    private final CrowdsecClientConfiguration crowdsecClientConfiguration;
+
+    @Inject
+    public CrowdsecImapConnectionCheck(CrowdsecClientConfiguration crowdsecClientConfiguration) {
+        this.crowdsecClientConfiguration = crowdsecClientConfiguration;
+    }
+
+    @Override
+    public Publisher<Void> validate(InetSocketAddress remoteAddress) {
+        String ip = remoteAddress.toString().split("/")[1].split(":")[0];
```

Comment:
This is way too hacky for me.

1/ split underneeth uses regexps and as such must be ban from production code. Use `Splitter.on('/')` instead.
2/ Extract the parsing of the IP into a dedicated method

```
String extractIp(InetSocketAddress address) {...}
```

And write dedicated tests for it.

3/ All this parsing ceremony is useless. Use third party APIs to do the job and read java doc. `address.getAddress().getHostAddress()` https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/net/InetSocketAddress.html#getAddress() https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/net/InetAddress.html#getHostAddress()